src/serviceWorkerRegistration.ts

```diff
@@ -24,18 +24,19 @@ type Config = {
 };
 
 export function register(config?: Config) {
-  if (process.env.NODE_ENV === 'production' && 'serviceWorker' in navigator) {
+  if (import.meta.env.PROD && 'serviceWorker' in navigator) {
     // The URL constructor is available in all browsers that support SW.
-    const publicUrl = new URL(process.env.PUBLIC_URL, window.location.href);
+    const publicUrl = new URL(import.meta.env.PUBLIC_URL, window.location.href);
     if (publicUrl.origin !== window.location.origin) {
       // Our service worker won't work if PUBLIC_URL is on a different origin
       // from what our page is served on. This might happen if a CDN is used to
       // serve assets; see https://github.com/facebook/create-react-app/issues/2374
+      // UPDATE 28/08/2024: Might be outdated since we moved from CRA to Vite
```

Comment:
The service worker registration seems broken now. I think we could try using `vite-plugin-pwa` instead. It can apparently generate the service worker automatically and seems less DYI than this CRA setup

Reply:
Sorry seems like I overlooked this. This seems to be fixed now?